File: codeforces/58.balanced_array/balanced_array.py
# Author: José Rodolfo (jric2002)
t = int(input())
for i in range(0, t):
  n = int(input())
  a = []
  if ((n % 4) == 0):
    j = 1
    while (j <= (n - 2)):
      a.append(j)
      j += 1
    a.append(j + 1)
    a.append(j + int(n / 2))
    print("YES")
    for k in range(1, (n - 2), 2):
      print(a[k], end=" ")
    print(a[n - 2], end=" ")
    for k in range(0, (n - 3), 2):
      print(a[k], end=" ")
    print(a[n - 1])
  else:
    print("NO")

File: codeforces/58.balanced_array/balanced_array.cpp
#include <iostream>
#include <vector>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned int t, n, j;
  vector<unsigned int> a;
  cin >> t;
  for (unsigned int i = 0; i < t; i++) {
    cin >> n;
    if ((n % 4) == 0) {
      j = 1;
      while (j <= (n - 2)) {
        a.push_back(j);
        j += 1;
      }
      a.push_back(j + 1);
      a.push_back(j + (n / 2));
      cout << "YES" << endl;
      for (unsigned int k = 1; k < (n - 2); k += 2) {
        cout << a.at(k) << " ";
      }
      cout << a.at(n - 2) << " ";
      for (unsigned int k = 0; k < (n - 3); k += 2) {
        cout << a.at(k) << " ";
      }
      cout << a.at(n - 1) << endl;
      a.clear();
    }
    else {
      cout << "NO" << endl;
    }
  }
  return 0;
}